javalib/src/main/scala/java/util/Collections.scala

```diff
@@ -0,0 +1,1131 @@
+package java.util
+
+import java.{lang => jl, io}
+
+import scala.annotation.tailrec
+import scala.language.implicitConversions
+import scala.collection.JavaConversions._
+
+object Collections {
+
+  import WrappedCollections._
+  import ImmutableCollections._
+  import CheckedCollections._
+
+  final lazy val EMPTY_SET: Set[_] = {
+    unmodifiableSet(
+      new AbstractSet[Any] with io.Serializable {
+        override def size(): Int = 0
+
+        override def iterator(): Iterator[Any] = emptyIterator[Any]
+      })
+  }
+
+  final lazy val EMPTY_LIST: List[_] = {
+    unmodifiableList(
+      new AbstractList[Any] with io.Serializable {
+        override def get(index: Int): Any =
+          throw new IndexOutOfBoundsException(index.toString)
+
+        override def size(): Int = 0
+      })
+  }
+
+  final lazy val EMPTY_MAP: Map[_, _] = {
+    unmodifiableMap[Any, Any](
+      new AbstractMap[Any, Any] with io.Serializable {
+        override def entrySet(): Set[Map.Entry[Any, Any]] =
+          EMPTY_SET.asInstanceOf[Set[Map.Entry[Any, Any]]]
+      })
+  }
+
+  def sort[T <: jl.Comparable[T]](list: List[T]): Unit =
+    sort(list, makeComparator[T])
+
+  def sort[T](list: List[T], c: Comparator[_ >: T]): Unit = {
+    val sortedList = list.sorted(c)
+    (0 until list.size).foreach(i => list(i) = sortedList(i))
```

Comment:
The second copying will be O(n²) if the list is a `LinkedList`. You should use `list.listIterator()`, and use its `set()` method to set the elements in O(n) time.